aws_marketplace/using-open-source-model-packages/pytorch-ic-model/notebook.ipynb

```diff
@@ -0,0 +1,461 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_name='image-classification-model'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#The image classification model packages this notebook notebook is compatible with, support application/x-image as the \n",
+    "#content-type.\n",
+    "content_type='application/x-image'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Review and update the compatible instance type for the model package in the following cell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "real_time_inference_instance_type='ml.g4dn.xlarge'\n",
+    "batch_transform_inference_instance_type='ml.p2.xlarge'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### A. Create an endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#create a deployable model from the model package.\n",
+    "model = ModelPackage(role=role,\n",
+    "                    model_package_arn=model_package_arn,\n",
+    "                    sagemaker_session=sagemaker_session)\n",
+    "\n",
+    "#Deploy the model\n",
+    "predictor = model.deploy(1, real_time_inference_instance_type, endpoint_name=model_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once endpoint has been created, you would be able to perform real-time inference."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### B. Prepare input file for performing real-time inference\n",
+    "In this step, we will download class_id_to_label_mapping from S3 bucket. The mapping files has been downloaded from [TensorFlow](https://storage.googleapis.com/download.tensorflow.org/data/ImageNetLabels.txt). [Apache 2.0 License](https://www.apache.org/licenses/LICENSE-2.0)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3_bucket = f\"jumpstart-cache-prod-{region}\"\n",
+    "key_prefix = \"inference-notebook-assets\"\n",
+    "\n",
+    "def download_from_s3(key_filenames):\n",
+    "    for key_filename in key_filenames:\n",
+    "        s3.download_file(s3_bucket, f\"{key_prefix}/{key_filename}\", key_filename)\n",
+    "\n",
+    "cat_jpg, dog_jpg, ImageNetLabels = \"cat.jpg\", \"dog.jpg\", \"ImageNetLabels.txt\"\n",
+    "\n",
+    "#Download images and label-mapping file.\n",
+    "download_from_s3(key_filenames=[cat_jpg, dog_jpg, ImageNetLabels])\n",
+    "\n",
+    "HTML('<table><tr><td> <img src=\"cat.jpg\" alt=\"cat\" style=\"height: 250px;\"/> <figcaption>cat.jpg</figcaption>'\n",
```

Comment:
Use an alternative method for display of images. HTML module has known compatibility issues.